framework/include/userobject/UserObjectInterface.h

```diff
@@ -84,14 +87,16 @@ template<class T>
 const T &
 UserObjectInterface::getUserObject(const std::string & name)
 {
-  return _uoi_feproblem.getUserObject<T>(_uoi_params.get<UserObjectName>(name));
+  unsigned int tid = isLocalUserObject(getUserObjectBase(name)) ? _uoi_tid : 0;
```

Comment:
Is any of that logic in the warehouse? Anyhow. These internals could be changed without changing the user facing interface.